csharp/Microsoft.Azure.Databricks.Client/UnityCatalog/SharesApiClient.cs

```diff
@@ -0,0 +1,77 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Text.Json;
+using System.Text.Json.Nodes;
+using System.Threading;
+using System.Threading.Tasks;
+
+using Microsoft.Azure.Databricks.Client.Models.UnityCatalog;
+using Microsoft.Azure.Databricks.Client.UnityCatalog.Interfaces;
+
+namespace Microsoft.Azure.Databricks.Client.UnityCatalog;
+
+public class SharesApiClient : ApiClient, ISharesApi
+{
+    public SharesApiClient(HttpClient httpClient) : base(httpClient)
+    {
+    }
+
+    public async Task<IEnumerable<Share>> List(CancellationToken cancellationToken = default)
```

Comment:
Looks like the REST api supports next_page_token. Can you support the ListPageable interface as well? See [here](https://github.com/Azure/azure-databricks-client/blob/a105775e8cb9faef49df48b6a7bbbf6959dfa091/csharp/Microsoft.Azure.Databricks.Client/PipelinesApiClient.cs#L20-L47) as a very similar example.